### PR TITLE
CLI and Release Pipeline Updates for configurable pipelines and templates #3498

### PR DESCRIPTION
**Pull request by mikefactorial**
Fixes https://github.com/microsoft/coe-starter-kit/issues/2170
Fixes https://github.com/microsoft/coe-starter-kit/issues/3495

### CODE REVIEW

File: ALMAcceleratorSampleSolution/deploy-test-ALMAcceleratorSampleSolution.yml [hunk 1]
```diff
@@ -6,12 +6,18 @@ resources:
     ref: main # If your pipeline templates are in a branch other than the default branch specify the branch here. Otherwise the default branch will be used by default.
     endpoint: coe-starter-kit # Specifies the GitHub service connection for this setup.
     name: Microsoft/coe-alm-accelerator-templates  #This is the name of the repo in the current project in Azure Devops that has the pipeline templates. If the repo is in a different project you can specify the project and repo using the format ProjectContainingTheBuildTemplates/RepositoryContainingTheBuildTemplates (https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema?view=azure-devops&tabs=schema%2Cparameter-schema#type)
+  - repository: PipelineRepo  # repository name (DO NOT CHANGE THIS VALUE)
```
Comment (mikefactorial) [inline ALMAcceleratorSampleSolution/deploy-test-ALMAcceleratorSampleSolution.yml:9]: @mikefactorial check this

File: coe-cli/src/commands/devops.ts [hunk 2]
```diff
@@ -32,6 +32,9 @@ import { InstalledExtension } from "azure-devops-node-api/interfaces/ExtensionMa
 
 import url from 'url';
 import { RoleAssignment } from "azure-devops-node-api/interfaces/SecurityRolesInterfaces";
+import { setTokenSourceMapRange, StringMappingType } from "typescript";
+import { pipeline } from "stream";
```
Comment (mikefactorial) [inline coe-cli/src/commands/devops.ts:36]: @mikefactorial is this necessary?

File: coe-cli/src/commands/devops.ts [hunk 3]
```diff
@@ -456,106 +465,109 @@ class DevOpsCommand {
         }
     }
 
-    async createMakersBuildVariables(args: DevOpsInstallArguments, connection: azdev.WebApi, securityContext: DevOpsProjectSecurityContext) {
-        connection = await this.createConnectionIfExists(args, connection)
+    async createMakersBuildVariables(args: DevOpsInstallArguments, connection: azdev.WebApi, securityContext: DevOpsProjectSecurityContext) 
+    {
+        let projects = [args.projectName, args.pipelineProjectName]
```
Comment (mikefactorial) [inline coe-cli/src/commands/devops.ts:470]: @mikefactorial need to check if args.projectName and args.pipelineProjectName are the same.

File: coe-cli/src/commands/devops.ts [hunk 4]
```diff
@@ -456,106 +465,109 @@ class DevOpsCommand {
...
-            }
+                let aadHost = Environment.getAzureADAuthEndpoint(aadArgs.settings).replace("https://", "")
+                if (!aadArgs.createSecret) {
+                    this.logger?.warn('Client secret not added for variable group alm-accelerator-variable-group it wil need to be added manually')
+                }
+
+                let buildApi = await connection.getBuildApi();
+                let builds = await buildApi.getDefinitions(projects[i])
+                let exportBuild = builds.filter(b => b.name == "export-solution-to-git")
+                let buildId = exportBuild.length == 1 ? exportBuild[0].id.toString() : ""
```
Comment (mikefactorial) [inline coe-cli/src/commands/devops.ts:503]: @mikefactorial What is this used for

**Review by jenschristianschroder — APPROVED** [review]

**Pull request merged**